pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation.go

```diff
@@ -5,337 +5,44 @@
-	instrumentationInstallTime := os.Getenv(instrumentationInstallTimeEnvVarName)
-	if instrumentationInstallTime == "" {
-		instrumentationInstallTime = common.ClusterAgentStartTime
-	}
-
-	return instrumentationInstallTime
-}
-
-// containerMutator creates a containerMutator for
-// telemetry environment variables pertaining to:
-//
-// - installation_time
-// - install_id
-// - injection_type
-func (s libInfoSource) containerMutator() containerMutator {
-	return containerMutators{
-		// inject DD_INSTRUMENTATION_INSTALL_TIME with current Unix time
-		envVarMutator(corev1.EnvVar{
-			Name:  instrumentationInstallTimeEnvVarName,
-			Value: s.instrumentationInstallTime(),
-		}),
-		// inject DD_INSTRUMENTATION_INSTALL_ID with UUID created during the Agent install time
-		envVarMutator(corev1.EnvVar{
-			Name:  instrumentationInstallIDEnvVarName,
-			Value: os.Getenv(instrumentationInstallIDEnvVarName),
-		}),
-		envVarMutator(corev1.EnvVar{
-			Name:  instrumentationInstallTypeEnvVarName,
-			Value: s.injectionType(),
-		}),
-	}
-}
-
-type extractedPodLibInfo struct {
-	// libs are the libraries we are going to attempt to inject into the given pod.
-	libs []libInfo
-	// languageDetection is set when we ran/used the language-detection annotation.
-	languageDetection *libInfoLanguageDetection
-	// source is where we got the data from, used for telemetry later.
-	source libInfoSource
-}
-
-func (e extractedPodLibInfo) withLibs(l []libInfo) extractedPodLibInfo {
-	e.libs = l
-	return e
-}
-
-func (e extractedPodLibInfo) useLanguageDetectionLibs() (extractedPodLibInfo, bool) {
-	if e.languageDetection != nil && len(e.languageDetection.libs) > 0 && e.languageDetection.injectionEnabled {
-		e.libs = e.languageDetection.libs
-		e.source = libInfoSourceSingleStepLangaugeDetection
-		return e, true
-	}
-
-	return e, false
-}
-
-func initContainerIsSidecar(container *corev1.Container) bool {
-	return container.RestartPolicy != nil && *container.RestartPolicy == corev1.ContainerRestartPolicyAlways
-}
-
-// podSumRessourceRequirements computes the sum of cpu/memory necessary for the whole pod.
-// This is computed as max(max(initContainer resources), sum(container resources) + sum(sidecar containers))
-// for both limit and request
-// https://kubernetes.io/docs/concepts/workloads/pods/sidecar-containers/#resource-sharing-within-containers
-func podSumRessourceRequirements(pod *corev1.Pod) corev1.ResourceRequirements {
-	ressourceRequirement := corev1.ResourceRequirements{
-		Limits:   corev1.ResourceList{},
-		Requests: corev1.ResourceList{},
+// NewAutoInstrumentation is a helper function to create a fully initialized webhook for SSI. Our webhook is made up of
+// several components, but consumers of this webhook should not need to care about how the webhook is wired together.
+func NewAutoInstrumentation(datadogConfig config.Component, wmeta workloadmeta.Component, rcClient *rcclient.Client) (*Webhook, error) {
```

Comment:
nit: Could we add "webhook" somewhere in the name? Makes it clear this is creating the webhook and not something else like the mutator.